kqueen/blueprints/api/views.py

```diff
@@ -31,86 +55,307 @@ def index():
 
 
 # Clusters
+
 @api.route('/clusters', methods=['GET'])
 @jwt_required()
 def cluster_list():
-    # TODO: implement native serialization
-
     output = []
 
-    for cluster in list(Cluster.list(return_objects=True).values()):
-        output.append(cluster.get_dict())
+    for obj in list(Cluster.list(return_objects=True).values()):
+        output.append(obj.get_dict())
+
+    return jsonify(output)
+
+
+@api.route('/clusters', methods=['POST'])
+@jwt_required()
+def cluster_create():
+    if not request.json:
+        abort(400)
+    else:
+        obj = Cluster(**request.json)
+        try:
+            obj.save()
+            output = obj.serialize()
```

Comment:
You should also call obj.engine.provision() to start provisioning of the cluster